### PR TITLE
Improve "Avoid using redundant else fixer"

### PR DESCRIPTION
**Pull request by louis-z**
- Improve `AvoidUsingRedundantElseFixer`
- Refactor tests a bit

### CODE REVIEW

File: src/Meziantou.Analyzer/Rules/AvoidUsingRedundantElseFixer.cs [hunk 1]
```diff
@@ -38,75 +38,46 @@ public override async Task RegisterCodeFixesAsync(CodeFixContext context)
...
+            var ifStatementParent = ifStatementSyntax.Parent;
+            if (ifStatementParent is null)
+                return document;
 
-            // Remove else
-            root = root.ReplaceNode(ifStatement, ifStatement.WithElse(@else: null));
-            ifStatement = (IfStatementSyntax)root.FindNode(ifStatement.IfKeyword.Span);
+            var modifiedIfStatement = ifStatementSyntax
+                .WithElse(null)
+                .WithTrailingTrivia(ifStatementSyntax.GetTrailingTrivia().Add(LineFeed));
```
Comment (meziantou) [inline src/Meziantou.Analyzer/Rules/AvoidUsingRedundantElseFixer.cs]: Could you add a test with an empty else statement

````
void A()
{
    if (true)
    {
         return;
    }
    else
    {
    }
}
````

Should fix with the following code (no empty line after the if)

````
void A()
{
    if (true)
    {
         return;
    }
}
````

Comment (louis-z) [reply src/Meziantou.Analyzer/Rules/AvoidUsingRedundantElseFixer.cs]: Good idea! It turns out the fixer was still outputting the empty line after the `if` block. 🙄 
Fixing it...
